### PR TITLE
Add gcc to muslPath

### PR DESCRIPTION
**Pull request by borkdude**
@thiagokokada I'm not sure if this is the correct fix, but it made the static musl build work for me.

### CODE REVIEW

File: mkGraal.nix [hunk 1]
```diff
@@ -48,7 +48,7 @@ let
     ++ lib.optionals gtkSupport [ cairo glib gtk3 ];
 
   muslPath = lib.makeBinPath [
-    musl.dev
+    musl.dev gcc
```
Comment (thiagokokada) [inline mkGraal.nix:51]: It must be `stdenv.cc` in this case, to ensure that we are using the same GCC as the current `stdenv` used during build.

Comment (thiagokokada) [reply mkGraal.nix:51]: Huh... Wait, this is macOS. macOS may be using `stdenv.cc` as `Clang` instead.

Comment (borkdude) [reply mkGraal.nix:51]: I wasn't testing this on macOS. I was testing on Debian + nix.

Comment (borkdude) [reply mkGraal.nix:51]: I can test with stdenv.cc

**Comment by thiagokokada** [comment]
This should fix it (I hope at least): https://github.com/thiagokokada/graalvm-nix/commit/a5e8162756514318791ed38480da02d808b84d99

**Comment by borkdude** [comment]
OK, will test

**Comment by borkdude** [comment]
@thiagokokada Confirming this works!

```
nix run "github:thiagokokada/graalvm-nix/a5e8162756514318791ed38480da02d808b84d99/#11/musl/21_3_0/native-image" -- --static --libc=musl Test
```

**Pull request closed without merging**